### PR TITLE
Upgrade jsx-a11y to 6.0.2, enable more jsx-a11y rules

### PR DESCRIPTION
**Pull request by wyattdanger**
Upgrades eslint-plugin-jsx-a11y from 5 to 6, and enables more a11y related rules

### CODE REVIEW

File: packages/eslint-config-airbnb/rules/react-a11y.js [hunk 1]
```diff
@@ -32,10 +32,6 @@ module.exports = {
     // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/aria-unsupported-elements.md
     'jsx-a11y/aria-unsupported-elements': 'error',
 
-    // disallow href "#"
-    // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/href-no-hash.md
```
Comment (wyattdanger) [inline packages/eslint-config-airbnb/rules/react-a11y.js]: this rule was removed in v6

File: packages/eslint-config-airbnb/rules/react-a11y.js [hunk 2]
```diff
@@ -55,9 +51,8 @@ module.exports = {
     'jsx-a11y/label-has-for': ['error', { components: ['label'] }],
 
     // require that mouseover/out come with focus/blur, for keyboard-only users
-    // TODO: evaluate
     // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/mouse-events-have-key-events.md
-    'jsx-a11y/mouse-events-have-key-events': 'off',
+    'jsx-a11y/mouse-events-have-key-events': 'error',
```
Comment (wyattdanger) [inline packages/eslint-config-airbnb/rules/react-a11y.js]: we already enable this in our codebase

File: packages/eslint-config-airbnb/rules/react-a11y.js [hunk 3]
```diff
@@ -110,7 +105,7 @@ module.exports = {
     // require onClick be accompanied by onKeyUp/onKeyDown/onKeyPress
     // https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/click-events-have-key-events.md
     // TODO: enable?
-    'jsx-a11y/click-events-have-key-events': 'off',
+    'jsx-a11y/click-events-have-key-events': 'error',
```
Comment (wyattdanger) [inline packages/eslint-config-airbnb/rules/react-a11y.js]: we already enable this in our codebase

**Comment by ljharb** [comment]
The airbnb config won't be compatible with v6 of jsx-a11y for awhile; even when we're compatible with eslint v4, we'll still be stuck on v5 for a bit. (#1447)

**Comment by wyattdanger** [comment]
Closing until later then

**Comment by kelset** [comment]
Any update on this, now that `15.1.0` is out?

**Comment by ljharb** [comment]
@kelset it will be included in the next semver-major of eslint-config-airbnb, but i want the latest release to sit for awhile first.

**Comment by justinhp** [comment]
@ljharb when will the next semver-major release be released?  It's just that it's preventing us from upgrading eslint-plugin-jsx-a11y which in turn makes us unable to upgrade eslint.  If it is going to be awhile longer then I can always fork this, no worries there.

**Comment by ljharb** [comment]
@justinhp i'm not sure what you mean; eslint-plugin-jsx-a11y v5 (which this config requires) works fine with any v3 or v4 eslint - so you shouldn't need to upgrade the a11y plugin in order to upgrade eslint.

**Comment by justinhp** [comment]
@ljharb thank you!  I'm an idiot.  I must have gotten confused with the various configs we have.  appreciate the help!

**Comment by ryansully** [comment]
Once peerDependencies has been updated, will we be able to install `eslint-config-airbnb` with a simple `npm install` or `yarn add` command, rather than the instructions in the package's README?

**Comment by ljharb** [comment]
@ryansully no, that's just the way peer deps work (requiring the install instructions) and that won't ever change until eslint itself comes up with a solution.

**Comment by scinos** [comment]
What is the status of this? when will the next semver-major release be released?

**Comment by ljharb** [comment]
@scinos status: not yet.

**Comment by inoas** [comment]
So when will this go live?

**Comment by ljharb** [comment]
@inoas same answer: not yet.

**Comment by inoas** [comment]
Whoot, but now the answer is ~~not~~ *yet* ;-)? <3

**Pull request merged**